### PR TITLE
Use flashinfer fused moe kernel path

### PR DESCRIPTION
**Pull request by guoqingbao**

### CODE REVIEW

**Comment by sempervictus** [comment]
Threw up a `flashinfer_moe/precision_flags` branch in my attn.rs fork - building without that first

**Comment by sempervictus** [comment]
On the 80B coder (FP8) not _much_ of a difference for either case at this point (the precision flags dont seem to hurt anymore) but it didn't explode... which is good.
```rust
2026-03-06T09:39:50.059070Z  INFO vllm_rs::core::block_manager: Prefix cache miss seq 2 (5432 tokens, 9 cached blocks, raw_match=0 blocks)
2026-03-06T09:39:51.010429Z  INFO vllm_rs::core::runner: User's thinking preference for reasoning models: None
2026-03-06T09:39:51.010446Z  WARN vllm_rs::core::runner: Using sampling from generation_config: temp=Some(1.0), top_k=Some(40), top_p=Some(0.95), freq_penalty=None, pres_penalty=None
2026-03-06T09:39:51.011484Z  INFO vllm_rs::utils::guidance: [llg] GuidanceState created successfully for grammar
2026-03-06T09:39:51.011488Z  INFO vllm_rs::utils::guidance: [llg] GuidanceState created successfully for grammar
2026-03-06T09:39:51.011491Z  INFO vllm_rs::utils::guidance: [llg] GuidanceState created successfully for grammar
2026-03-06T09:39:51.011569Z  INFO vllm_rs::utils::guidance: [llg] GuidanceState created successfully for grammar
2026-03-06T09:39:51.020406Z  INFO vllm_rs::core::engine: Prefilling [seq_id 2]: 5433 tokens in 0.99s (5476.81 tokens/s)
2026-03-06T09:39:56.038745Z  INFO vllm_rs::core::engine: Decoding: 1 active request(s) [Seq: [2]], avg. 53 tokens/s per request (total: 53 tokens/s)
2026-03-06T09:39:56.038781Z  INFO vllm_rs::core::scheduler: GPU Kvcache: 60944 blocks (3900416 tokens) free, used 7.0% (3.36GB/48.00GB); CPU swap used 0.0% (0.00GB/96.00GB)
2026-03-06T09:39:56.038792Z  INFO vllm_rs::core::scheduler: GPU MambaState: 1 / 377 slots used (0.3%), approx 0.02GB/6.79GB (slot 18.42MB)
2026-03-06T09:40:01.047788Z  INFO vllm_rs::core::engine: Decoding: 1 active request(s) [Seq: [2]], avg. 53 tokens/s per request (total: 53 tokens/s)
2026-03-06T09:40:06.058474Z  INFO vllm_rs::core::engine: Decoding: 1 active request(s) [Seq: [2]], avg. 53 tokens/s per request (total: 53 tokens/s)
2026-03-06T09:40:11.070231Z  INFO vllm_rs::core::engine: Decoding: 1 active request(s) [Seq: [2]], avg. 53 tokens/s per request (total: 53 tokens/s)
2026-03-06T09:40:11.070272Z  INFO vllm_rs::core::scheduler: GPU Kvcache: 60932 blocks (3899648 tokens) free, used 7.0% (3.37GB/48.00GB); CPU swap used 0.0% (0.00GB/96.00GB)
2026-03-06T09:40:11.070284Z  INFO vllm_rs::core::scheduler: GPU MambaState: 1 / 377 slots used (0.3%), approx 0.02GB/6.79GB (slot 18.42MB)
2026-03-06T09:40:16.079026Z  INFO vllm_rs::core::engine: Decoding: 1 active request(s) [Seq: [2]], avg. 53 tokens/s per request (total: 53 tokens/s)
2026-03-06T09:40:16.585547Z  INFO vllm_rs::core::block_manager: Prefix cache insert seq 2 (6794 tokens, 106 blocks)
2026-03-06T09:40:16.585630Z  WARN vllm_rs::server::server: --- Performance Metrics ---
2026-03-06T09:40:16.585660Z  INFO vllm_rs::server::server: [Seq 2] ⏱️ Prompt: 5432 tokens in 0.99s (5475.81 t/s)
2026-03-06T09:40:16.585670Z  INFO vllm_rs::server::server: [Seq 2] ⏱️ Decoded: 1362 tokens in 25.57s (53.28 t/s)
2026-03-06T09:40:16.589960Z  INFO vllm_rs::core::scheduler: GPU Kvcache: 60928 blocks (3899392 tokens) free, used 7.0% (3.38GB/48.00GB); CPU swap used 0.0% (0.00GB/96.00GB)
2026-03-06T09:40:16.589969Z  INFO vllm_rs::core::scheduler: GPU MambaState: 1 / 377 slots used (0.3%), approx 0.02GB/6.79GB (slot 18.42MB)
```
we can get some more decode speed by constraining the output logits using something like `((?s)[\x00-\u{10FFFF}])` as the mask but for the time being, "it all works together" (this has #232 in it)

**Comment by sempervictus** [comment]
Interesting: https://github.com/flashinfer-ai/flashinfer/tree/65d6e4a2e104cbe963856cdf39cd363ee7c74a1f/include/flashinfer/mamba
